code/datums/components/multitile.dm

```diff
@@ -16,45 +16,64 @@
  */
 
 //distance_from_center does not include src itself
-/datum/component/multitile/Initialize(distance_from_center, new_filler_map)
-	if(!distance_from_center || !length(new_filler_map))
+/datum/component/multitile/Initialize(new_filler_map)
+	if(!length(new_filler_map))
 		return COMPONENT_INCOMPATIBLE
 
 	if(!isatom(parent))
 		return COMPONENT_INCOMPATIBLE
+	var/max_height = length(new_filler_map)
+	var/max_width = length(new_filler_map[1]) //it should have the same length on every row
+
+	var/offset_x = 0
+	var/offset_y = 0
 
 	var/atom/owner = parent
-	if(owner.x + distance_from_center > world.maxx || owner.x - distance_from_center < 1)
+
+	for(var/i in 1 to length(new_filler_map))
+		if(length(new_filler_map[i] != max_width))
+			stack_trace("A multitile component was passed a list wich did not have the same length every row. Atom parent is: [parent]")
+			var/obj/machinery/machine = parent
+			machine.deconstruct()
+			return COMPONENT_INCOMPATIBLE
+
+		for(var/j in 1 to length(new_filler_map[i]))
+			if(new_filler_map[i][j] == 2)
```

Comment:
```suggestion
			if(new_filler_map[i][j] == MACH_CENTER)
```
Am I understanding this code block correctly? If so this is better for readability

Reply:
done